src/context/node.js

```diff
@@ -403,11 +419,7 @@ export class ContextNode {
         // fast operation.
         for (let i = 0; i < parentChildren.length; i++) {
           const child = parentChildren[i];
-          if (
-            child != this &&
-            child.isDiscoverable() &&
-            this.node.contains(child.node)
-          ) {
+          if (child != this && child.isDiscoverable()) {
```

Comment:
can you explain this change?

Reply:
Yeah, not super happy about it, tbh. But I didn't see any other way. But shadow root "breaks" the `contains()` API a bit for us. For a slot `slot.contains(assignedNode)` is always `false` because they are in different subtrees. So `this.node.contains(child.node)` would prevent an assigned node to be rediscovered. This change will cause more rediscovers. However, the cost of trying to rediscover a node is relatively low. So the correctness of code won out for me. There's a way to do a more complicated `contains` over the boundaries of light/shadow subtrees. But that code is exactly what `discover` does. So this also avoids executing that code twice.